analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs

```diff
@@ -83,7 +106,62 @@ public void ByDefault_ExecuteRule()
                 Verifier.VerifyAnalyzer(testCase.Path,
                                         testCase.Analyzer,
                                         ParseOptionsHelper.FromCSharp8,
-                                        MetadataReferenceFacade.SystemComponentModelPrimitives);
+                                        ConcatReferences(testCase.AdditionalReferences));
+            }
+        }
+
+        [TestMethod]
+        public void WhenProjectType_IsTest_RunsOnlyTestRules()
+        {
+            var sonarProjectConfig = TestHelper.CreateSonarProjectConfig(Directory.CreateDirectory(@"TestCases\WhenProjectType_IsTest_RunsOnlyTestRules").FullName, ProjectType.Test);
+            foreach (var testCase in testCases)
+            {
+                var isTest = testCase.Analyzer.SupportedDiagnostics.Any(d => d.CustomTags.Contains(DiagnosticDescriptorBuilder.TestSourceScopeTag));
```

Comment:
`has` to describe the fact that it might be Main&Test and not exclusively Test.
`Scope` to differentiate between rule scope from RSPEC and compilation/scanner property.
```suggestion
                var hasTestScope = testCase.Analyzer.SupportedDiagnostics.Any(d => d.CustomTags.Contains(DiagnosticDescriptorBuilder.TestSourceScopeTag));
```